rll/automate_upgrade.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Add entry in /etc/cron.d/ to daily check and excute an upgrade for rightlink.
+
+# -e will immediatly exit out of script at point of error
+set -e
+
+cron_file='/etc/cron.d/rightlink_upgrade'
+exec_file='/usr/local/bin/rightlink_check_upgrade'
+
+# Grab toggle option to disable
+if [[ "$DISABLE_AUTO_UPGRADE" == 'true' ]]; then
+  if [ -e ${cron_file} ]; then
+    rm -f ${cron_file}
+    echo "Automatic upgrade disabled"
+  else
+    echo "Automatic upgrade never enabled - no action done"
+  fi
+else
+  # If cron file already exists, will recreate it and update cron config with new random times.
+  [ -e ${cron_file} ] && echo "Recreating cron entry"
+
+  # Generate executable script to run by cron
+  cat > ${exec_file} << 'EOF'
+#!/bin/bash
+
+# This file is autogenerated. Do not edit as changes will be overwritten.
+
+source /var/run/rll-secret
+curl --silent --show-error --get --globoff --fail --request POST --header X-RLL-Secret:$RS_RLL_SECRET \
+  http://127.0.0.1:$RS_RLL_PORT/rll/run/recipe --data-urlencode 'recipe=rll::upgrade'
+EOF
+  chown root:root ${exec_file}
+  chmod 0700 ${exec_file}
+
+  # Random hour 0-23
+  cron_hour=$[ $RANDOM % 24 ]
+
+  # Random minute 0-59
+  cron_minute=$[ $RANDOM % 60 ]
```

Comment:
This type of arithmetic expansion in Bash is deprecated, please use `$(( ... ))` instead.